### ISSUE TITLE
Destroy hangs if there is nothing to destroy

### ISSUE THREAD

**Issue by rparmer** [weaveworks/tf-controller#495]
I've got a terraform resource that has `destroyResourcesOnDeletion` set to true.  There was error in the values that prevented any resources from being created.  This was expected, however when I deleted it the resource got stuck in `Applying` with the following log message

```
No changes. No objects need to be destroyed.

Either you have not created any objects yet or the existing objects were
already deleted outside of Terraform.
```

Manually editing the resource finalizers fixed it.  But I'm guess this is an edge case status that needs to be checked.

**Comment by darrylweaver** [comment]
I can confirm this issue.
If there are no resources to delete, the deletion of the terraform resource hangs on the finalizer and you have to edit the resource directly to remove the finalizer so it finishes deletion.

To recreate:

I create a Terraform resource and set: 
```
spec:
  destroy: false
  destroyResourcesOnDeletion: true
```
The TF plan is generated and the apply deploys all the resources.
I then set 
```
spec:
  destroy: true
  destroyResourcesOnDeletion: true
```
The Terraform destroy plan is generated and applied deleting all resources.
Then delete the terraform resource and it hangs on deletion.
The output from the runner says:
```
No changes. No objects need to be destroyed.

Either you have not created any objects yet or the existing objects were
already deleted outside of Terraform.

```

Expected behaviour:

Ops/Infra/Platform teams want to be able to control deletion of resources with TF as this is typically risky and does not always delete resources correctly and could need some fixing of HCL code during testing.
I would expect to be able to set both "destroy" to provide control and "destroyResourcesOnDeletion" to avoid orphaned resources.
I should be able to destroy the resources in one step and delete the terraform resource in the next step.
The runner should complete successfully when running a destroy plan that has no resources.

**Comment by darrylweaver** [comment]
Using the latest Helm release 0.11.0 with image: ghcr.io/weaveworks/tf-controller:v0.14.0

**Comment by chanwit** [comment]
@darrylweaver it's fixed on the master branch.

Please use these images:

TF-controller:
`ghcr.io/weaveworks/tf-controller:main-be987f0b`
Runner:
`ghcr.io/weaveworks/tf-runner:main-be987f0b`

**Comment by darrylweaver** [comment]
I upgraded the TF-controller images to the version main-be987f0b
The tf-controller updated, correctly.
However, I now cannot see the plan in WGE UI for the existing TF resource on the cluster.
The UI page for the plan comes up with an error.

Page URL: 
[rds-database-demo](https://demo3.weavegitops.com/terraform/object/plan?clusterName=management&name=rds-database-demo&namespace=flux-system)
Error:
```
Something went wrong.
Cannot read properties of undefined (reading 'kind')
TypeError: Cannot read properties of undefined (reading 'kind')
    at gr (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:3673928)
    at oo (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4922386)
    at _c (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4974686)
    at wl (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4961927)
    at Cl (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4961855)
    at Ml (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4961718)
    at Al (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4958684)
    at https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4908178
    at t.unstable_runWithPriority (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:5684427)
    at Wa (https://demo3.weavegitops.com/static/js/main.214ad00a.js:2:4907955)
```

**Comment by chanwit** [comment]
@darrylweaver I checked in WGE and your chart was upgraded automatically to v0.12.0 which has breaking changes for Flux v2.0.0-rc.1
Please fix the Helm chart version to v0.11.0 to try if it's solved the UI.

**Comment by darrylweaver** [comment]
Downgrading to 0.11.0 doesn't work as it complains about the missing CRD version.
Reverting back to 0.12.0 and images 0.14.0 and the UI is still broken :(

**Comment by rparmer** [comment]
I'm seeing the same thing.  tf-controller 0.12.0, wge 0.21.2, flux v2.0.0-rc.1.  The error only shows up when trying to view the plan in the UI.

**Comment by chanwit** [comment]
Planned to release chart 0.11.1 to help fix this issue.
the new chart would ship both CRDs, but change back the storage version to v1alpha1.
I'll keep you posted.

**Comment by darrylweaver** [comment]
Update: the issue with the TF plan page in the UI was unrelated to this issue, this was fixed in WGE 0.22.0.

For this issue:
I tested the images:
TF-controller:
ghcr.io/weaveworks/tf-controller:main-be987f0b
Runner:
ghcr.io/weaveworks/tf-runner:main-be987f0b

with Helm chart version 0.11.0 on demo3.
and it will now delete successfully even when there is no resources to delete in the plan, so we can close this issue now.
